dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -318,73 +335,100 @@ public boolean doesUserHavePermission(final Permissionable permissionable, int p
 					throw new DotRuntimeException(e1.getMessage(), e1);
 				}
+
+				if (permissionable instanceof WorkflowAction
+					&& workflowActionHasPermission(expecterPermissionType, p, user, contentlet)) {
+					return true;
+				}
 			}
 		}
+
         // front end users cannot read content that is not live
-        if(!user.isBackendUser() && isContentlet && !isLiveContentlet(permissionable) && permissionType == PERMISSION_READ) {
+        if (!user.isBackendUser()
+			&& isContentlet
+			&& !isLiveContentlet(permissionable)
+			&& expecterPermissionType == PERMISSION_READ) {
             Logger.warn(this, String.format("User '%s' cannot verify READ permissions on Contentlet '%s' because it " +
                     "is not live.", user.getUserId(), permissionable.getPermissionId()));
             return false;
         }
 
+		final Set<Role> roles = new HashSet<>(Try
+			.of(() -> APILocator.getRoleAPI().loadRolesForUser(user.getUserId()))
+			.getOrElse(List.of()));
 
-		final Set<Role> roles = new HashSet<>(Try.of(()->APILocator.getRoleAPI().loadRolesForUser(user.getUserId())).getOrElse(new ArrayList<>()));
-        
-
-		
 		// remove front end user access for anon user (e.g, /intranet)
 		// Note to selves: it was a mistake to add this role to the Anon user in 5.2.0
-		if(user.isAnonymousUser()) {
+		if (user.isAnonymousUser()) {
             roles.remove(frontEndUserRole);
         }
-		if(!respectFrontendRoles) {
+		if (!respectFrontendRoles) {
 		    roles.remove(frontEndUserRole);
 		    roles.remove(anonRole);
 		    roles.remove(APILocator.getRoleAPI().loadRoleByKey("anonymous"));
 		}
-		
-		Set<String> userRoleIds= roles.stream().map(r->r.getId()).collect(Collectors.toSet());
-        
+		final Set<String> userRoleIds = roles.stream().map(Role::getId).collect(Collectors.toSet());
         
-		return doRolesHavePermission(userRoleIds,getPermissions(permissionable, true),permissionType);
+		return doRolesHavePermission(userRoleIds, getPermissions(permissionable, true), expecterPermissionType);
 	}
 
-    /**
+	/**
+	 * Given an expected permissionType to be satisfied, a resolved "Anyone who can..." style permission, a {@link User}
+	 * and a {@link Contentlet}, evaluates if the contentlet is new. Then verify that the provided permission is the
+	 * actual mentioned style and if it does then finally evaluate the contentlet's content type permissions as a
+	 * fallback.
+	 *
+	 * @param permissionType provided permission type
+	 * @param permission actual permission
+	 * @param user provided user
+	 * @param contentlet contentlet
+	 * @return true if the explained logic takes place, otherwise false
+	 */
+	private boolean workflowActionHasPermission(final int permissionType,
+												final Permission permission,
+												final User user,
+												final Contentlet contentlet) throws DotDataException {
+		if (Objects.nonNull(contentlet) && StringUtils.isBlank(contentlet.getInode())) {

```

Comment:
I am not sure if you need the contentlet here, since the Permissionable I think is actually the contentlet, so you can basically check if the permissionable is a Contentlet and cast to it

Reply:
@jdotcms I tried to. But in this case the `Permissionable` here is actually the `WorkflowAction`, not the contentlet.

Reply:
ok